src/encoded/search.py

```diff
@@ -1118,3 +1120,389 @@ def news(context, request):
+    aggs.update(update_temp)
+
+    # If internal action data is able to be seen in facets (if logged in) then add it to aggs for 
+    # both: 1) an audit category row for Internal Action and 2) the no audits row as innermost 
+    # level of nested query to create a no audits at all row within no audits.
+    # Additionally, add it to the no_audits_groupings list to be used in summarize_no_audits later.
+    if "audit.INTERNAL_ACTION.category" in facets[len(facets)-1]:
+        aggs['audit.INTERNAL_ACTION.category'] = {'aggs': {x_grouping: x_agg
+                },
+            'terms': {'field': 'audit.INTERNAL_ACTION.category', 'size': 0
+                    }
+            }
+        aggs['no.audit.error']['aggs']['no.audit.not_compliant']['aggs']['no.audit.warning']['aggs']['no.audit.internal_action'] = {
+                            "missing": {
+                                "field": "audit.INTERNAL_ACTION.category"
+                            },
+                            "aggs": {
+                                x_grouping: x_agg
+                            }
+                        }
+        no_audits_groupings.append("no.audit.internal_action")
+
+    aggs['x'] = x_agg
+    query['aggs']['matrix'] = {
+        "filter": {
+            "bool": {
+                "must": filters,
+            }
+        },
+        "aggs": aggs,
+    }
+
+    # Execute the query
+    es_results = es.search(body=query, index=es_index, search_type='count')
+
+    # Format matrix for results
+    aggregations = es_results['aggregations']
+    result['matrix']['doc_count'] = total = aggregations['matrix']['doc_count']
+    result['matrix']['max_cell_doc_count'] = 0
+
+    # Format facets for results
+    result['facets'] = format_facets(
+        es_results, facets, used_filters, (schema,), total, principals)
+
+    # For each audit category row.
+    # Convert Elasticsearch returned matrix search data to a form usable by the front end matrix
+    # code, using only the 'matrix' object within the search data. It contains matrix search 
+    # results in 'bucket' arrays, with the exact terms being defined in the .py file for the object
+    # we're querying in the object's 'matrix' property.
+    #
+    # matrix (dictionary): 'matrix' object within the Elasticsearch matrix search results,
+    #        containing all the data to render the matrix and its headers, but not the facets that
+    #        appear along the top and bottom.
+    # x_buckets (dictionary): 'x' object within the Elasticsearch matrix search results, containing
+    #        the headers that give titles to each column of the chart, as well as summary counts
+    #        we don't currently use on the front end.
+    # outer_bucket (list): search result bucket containing the term from which to group by.
+    # grouping_fields (list): List of audit categories that is hard coded above.
+    
+    def summarize_buckets(matrix, x_buckets, outer_bucket, grouping_fields):
```

Comment:
This comes from David, and has been in matrix since the beginning. I’m doing the same thing for target matrix.